drivers/include/periph/adc.h

```diff
@@ -87,6 +87,10 @@ typedef unsigned int adc_t;
 
 /**
  * @brief   Possible ADC resolution settings
+ *
+ * @details This type has to be provided by the underlying implementation if
+ *          the set of supported resolutions is different. Only resolutions
+ *          actually supported by the board must be defined.
```

Comment:
Since the resolutions are defined in an `enum`, we can't do `#ifdef ADC_RES_16BIT`, so how can we determine which resolutions are supported?

Either we have to change this to a `#define` or we need to come up with something else.

Reply:
> Either we have to change this to a `#define` or we need to come up with something else.

Might pseudomodules for each resolution or/and `FEATURES_PROVIDED += adc_res_12` help?

Reply:
After thinking a bit about it, I now think this is actually a bad idea. Sooner or later we will want to add support for external ADCs, or MCU with multiple heterogeneous ADCs.

Reply:
But on the other hand the hole API cannot be extended in any reasonable way to support heterogeneous ADCs. E.g. with having different values for the same item in `adc_res_t` depending on MCU makes this utterly complex.

Maybe it is time to propose a different ADC API. This API makes it not only unnecessary hard to support multiple ADCs, but also to expose features like selecting the reference voltage, performing multiple conversions into a buffer via DMA, keeping the ADC online for fast conversions, etc.

Reply:
> Maybe it is time to propose a different ADC API. This API makes it not only unnecessary hard to support multiple ADCs.

In PR #10504 I started a try to change the ADC API where an ADC line would then be defined as a channel of an ADC device. Each device might then have different capabilities and configurations.

My plans were to resume my work in PR #10527 for an ADC extension API once the GPIO extension API in PR #12877 is on a good way. But at the moment, I'm not working on it.

If you start to think about a new ADC API, please have the use of ADC extenders in mind.